### PR TITLE
chore: remove old (bugfix/migration) code

### PR DESCRIPTION
**Pull request by minisbett**
# Describe your changes

Removed various code that is either already old and commented out for a long time or presented updates for very old bancho.py times (e.g. when the country code wasn't saved yet) to clean up the codebase a bit.

## Checklist
- [X] I've manually tested my code
- [X] The changes pass pre-commit checks (`make lint`)
- [X] The changes follow coding style

### CODE REVIEW

File: app/objects/beatmap.py [hunk 1]
```diff
@@ -407,6 +408,7 @@ async def from_md5(cls, md5: str, set_id: int = -1) -> Beatmap | None:
     @classmethod
     async def from_bid(cls, bid: int) -> Beatmap | None:
         """Fetch a map from the cache, database, or osuapi by id."""
+        print(f"from_bid({bid})")
```
Comment (tsunyoku) [inline app/objects/beatmap.py:411]: ?

File: app/objects/beatmap.py [hunk 2]
```diff
@@ -362,6 +362,7 @@ def as_dict(self) -> dict[str, object]:
     @classmethod
     async def from_md5(cls, md5: str, set_id: int = -1) -> Beatmap | None:
         """Fetch a map from the cache, database, or osuapi by md5."""
+        print(f"from_md5({md5})")
```
Comment (tsunyoku) [inline app/objects/beatmap.py:365]: ?

File: app/objects/beatmap.py [hunk 3]
```diff
@@ -495,8 +497,11 @@ def _parse_from_osuapi_resp(self, osuapi_resp: dict[str, Any]) -> None:
         # if a map is 'frozen', we keep its status
         # even after an update from the osu!api.
         if not getattr(self, "frozen", False):
+            print("FROZEN IS NOT TRUE ON " + self.__repr__())
```
Comment (tsunyoku) [inline app/objects/beatmap.py:500]: ?

File: app/objects/beatmap.py [hunk 4]
```diff
@@ -495,8 +497,11 @@ def _parse_from_osuapi_resp(self, osuapi_resp: dict[str, Any]) -> None:
...
             osuapi_status = int(osuapi_resp["approved"])
             self.status = RankedStatus.from_osuapi(osuapi_status)
+        else:
+            print("FROZEN ON " + self.__repr__())
```
Comment (tsunyoku) [inline app/objects/beatmap.py:504]: ?

File: app/objects/beatmap.py [hunk 5]
```diff
@@ -601,6 +606,7 @@ def _cache_expired(self) -> bool:
         """Whether the cached version of the set is
         expired and needs an update from the osu!api."""
         current_datetime = datetime.now()
+        return True
```
Comment (tsunyoku) [inline app/objects/beatmap.py:609]: ?

File: app/objects/beatmap.py [hunk 6]
```diff
@@ -667,13 +673,20 @@ async def _update_if_available(self) -> None:
...
                         int(new_map["approved"]),
                     )
+                    print(f"checking map that is frozen={old_map.frozen}")
                     if (
                         old_map.md5 != new_map["file_md5"]
                         or old_map.status != new_ranked_status
                     ):
+                        print(
+                            f"update because {old_map.md5} != {new_map['file_md5']} or {old_map.status} != {new_ranked_status}",
+                        )
```
Comment (tsunyoku) [inline app/objects/beatmap.py:683]: ?

File: app/objects/beatmap.py [hunk 7]
```diff
@@ -667,13 +673,20 @@ async def _update_if_available(self) -> None:
                     new_ranked_status = RankedStatus.from_osuapi(
                         int(new_map["approved"]),
                     )
+                    print(f"checking map that is frozen={old_map.frozen}")
```
Comment (tsunyoku) [inline app/objects/beatmap.py:676]: ?

File: app/objects/beatmap.py [hunk 8]
```diff
@@ -667,13 +673,20 @@ async def _update_if_available(self) -> None:
...
                     ):
+                        print(
+                            f"update because {old_map.md5} != {new_map['file_md5']} or {old_map.status} != {new_ranked_status}",
+                        )
                         # update map from old_maps
                         bmap = old_maps[old_id]
                         bmap._parse_from_osuapi_resp(new_map)
+                        print(
+                            f"beatmap status after _parse_from_osuapi_resp with new_ranked_status being {new_ranked_status}: {bmap.status}",
+                        )
```
Comment (tsunyoku) [inline app/objects/beatmap.py:689]: ?

File: app/objects/beatmap.py [hunk 9]
```diff
@@ -690,14 +703,19 @@ async def _update_if_available(self) -> None:
 
                     # (some implementation-specific stuff not given by api)
                     bmap.frozen = False
+                    print("FROZEN WAS SET TO FALSE BECAUSE NEW MAP " + bmap.__repr__())
+                    print(new_map)
+                    print(old_maps)
```
Comment (tsunyoku) [inline app/objects/beatmap.py:708]: ?

File: app/objects/beatmap.py [hunk 10]
```diff
@@ -690,14 +703,19 @@ async def _update_if_available(self) -> None:
...
                     bmap.passes = 0
                     bmap.plays = 0
 
                     bmap.set = self
                     updated_maps.append(bmap)
 
             # save changes to cache
+            print(f"length of self.maps before: {len(self.maps)}")
```
Comment (tsunyoku) [inline app/objects/beatmap.py:716]: ?

File: app/objects/beatmap.py [hunk 11]
```diff
@@ -690,14 +703,19 @@ async def _update_if_available(self) -> None:
...
             self.maps = updated_maps
+            print(f"length of self.maps after: {len(self.maps)}")
```
Comment (tsunyoku) [inline app/objects/beatmap.py:718]: ?

File: app/objects/beatmap.py [hunk 12]
```diff
@@ -903,6 +905,8 @@ async def _from_bsid_osuapi(cls, bsid: int) -> BeatmapSet | None:
                 else:
                     bmap.frozen = False
 
+                print("FROZEN ON THE NEW MAP WAS SET TO " + bmap.frozen.__repr__())
+
```
Comment (tsunyoku) [inline app/objects/beatmap.py:909]: ?

File: app/objects/beatmap.py [hunk 13]
```diff
@@ -932,6 +936,7 @@ async def _from_bsid_osuapi(cls, bsid: int) -> BeatmapSet | None:
     async def from_bsid(cls, bsid: int) -> BeatmapSet | None:
         """Cache all maps in a set from the osuapi, optionally
         returning beatmaps by their md5 or id."""
+        print(f"from_bsid({bsid})")
```
Comment (tsunyoku) [inline app/objects/beatmap.py:939]: ?

**Review by tsunyoku — CHANGES_REQUESTED** [review]

File: app/objects/beatmap.py (hunk 1 above)
Comment (minisbett) [reply app/objects/beatmap.py:411]: oh no..

File: app/objects/beatmap.py (hunk 2 above)
Comment (minisbett) [reply app/objects/beatmap.py:365]: htfxhnhfxtmm, x

File: app/objects/beatmap.py (hunk 3 above)
Comment (minisbett) [reply app/objects/beatmap.py:500]: txu ,u6trxr8

File: app/objects/beatmap.py (hunk 4 above)
Comment (minisbett) [reply app/objects/beatmap.py:504]: 5t{]  568 54 eyd5r7,u5

File: app/objects/beatmap.py (hunk 5 above)
Comment (minisbett) [reply app/objects/beatmap.py:609]: y, 8etbethse

File: app/objects/beatmap.py (hunk 6 above)
Comment (minisbett) [reply app/objects/beatmap.py:683]: drz erzue<rzumz

File: app/objects/beatmap.py (hunk 7 above)
Comment (minisbett) [reply app/objects/beatmap.py:676]: n z5e5ze5m

File: app/objects/beatmap.py (hunk 8 above)
Comment (minisbett) [reply app/objects/beatmap.py:689]: vw3r<t3vw4tvw34b<

File: app/objects/beatmap.py (hunk 9 above)
Comment (minisbett) [reply app/objects/beatmap.py:708]: , zi7of,io7 zi7,

File: app/objects/beatmap.py (hunk 10 above)
Comment (minisbett) [reply app/objects/beatmap.py:716]: nr5uz ymu5r y5ury

File: app/objects/beatmap.py (hunk 11 above)
Comment (minisbett) [reply app/objects/beatmap.py:718]: bze45 ybze45 zbe45

File: app/objects/beatmap.py (hunk 13 above)
Comment (minisbett) [reply app/objects/beatmap.py:939]: fgxhnxf jxr56,l

File: app/objects/beatmap.py (hunk 12 above)
Comment (minisbett) [reply app/objects/beatmap.py:909]: hery5ze5ze4a5a

File: app/api/domains/cho.py [hunk 14]
```diff
@@ -787,19 +787,6 @@ async def login(
...
-        log(f"Fixing {login_data['username']}'s country.", Ansi.LGREEN)
-
-        await db_conn.execute(
-            "UPDATE users SET country = :country WHERE id = :user_id",
-            {
-                "country": geoloc["country"]["acronym"],
-                "user_id": user_info["id"],
-            },
-        )
-
```
Comment (cmyui) [inline app/api/domains/cho.py:802]: this is likely still in use on some older bancho.py installs, so would prefer to keep it. low risk/reward ratio

File: app/commands.py [hunk 15]
```diff
@@ -339,15 +339,8 @@ async def recent(ctx: Context) -> str | None:
...
     else:
-        # XXX: prior to v3.2.0, bancho.py didn't parse total_length from
-        # the osu!api, and thus this can do some zerodivision moments.
-        # this can probably be removed in the future, or better yet
-        # replaced with a better system to fix the maps.
-        if score.bmap.total_length != 0:
-            completion = score.time_elapsed / (score.bmap.total_length * 1000)
-            l.append(f"FAIL {{{completion * 100:.2f}% complete}})")
-        else:
-            l.append("FAIL")
```
Comment (cmyui) [inline app/commands.py:350]: migrating away from this would require a script to fix people's `total_length` in the case they're 0

**Comment by cmyui** [comment]
i generally think prs removing backwards compatibilities are tricky to merge, it's kinda risky for low reward

File: app/api/domains/cho.py (hunk 14 above)
Comment (minisbett) [reply app/api/domains/cho.py:802]: I doubt 2 year old migrations are still relevant. I think we should define a baseline on how long migrations are maintained and comment that inside the code. I think more than 6 months is already a lot, users really should update bpy within 6 months. Also considering all those changes that happened recently I doubt someone is trying to migrate some 2 year old server

File: app/commands.py (hunk 15 above)
Comment (minisbett) [reply app/commands.py:350]: This fix is also *really" old, I don't think any maps with 0 length still exist

**Comment by cmyui** [comment]
given they're pretty minor but can result in some pretty bad states i think i'm more comfortable leaving these for now.

i think i would be open to accepting a well tested script to backfill the data issues, at which point i would be willing to remove the jit data fixes here

**Pull request closed without merging**